plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -777,6 +778,19 @@ public HiveConfig setSortedWritingEnabled(boolean sortedWritingEnabled)
         return this;
     }
 
+    public boolean isOptimizeSortedFileReads()
+    {
+        return optimizeSortedFileReads;
+    }
+
+    @Config("hive.optimize-sorted-file-reads")
+    @ConfigDescription("Enable optimizations for reading sorted files")
```

Comment:
Can the name be more descriptive?

I understand the value in having a broad config name, as we can stuff more features without need for renames or new toggles, but otoh the current toggle name doesn't tell me much.

If enabling the toggle can affect correctness, it should be indicated in the description and/or naming

Reply:
I am planning on documenting what the toggle does and its caveats in the user facing docs.
I think correctness issues should only arise if a user generates files directly and defines sorted_by on the table without actually sorting the data as defined in DDL. In typical scenario, engines should enforce sorting on writes to sorted tables. This seems to be similar to the caveats which apply when `hive.bucket-execution` is used.
I've updated the description to `Use sorted table layout to generate more efficient execution plans. May lead to incorrect results if files are not sorted as per table definition.`.
I'm open to suggestions on toggle name, it seems tricky to come up with something more descriptive without making it too verbose or specific to current code.